packages/components/src/assay/AssayButtons.tsx

```diff
@@ -0,0 +1,238 @@
+    const onDeleteRun = useCallback(() => {
+        setShowConfirmDelete(true);
+    }, []);
+
+    const afterDelete = useCallback(() => {
+        onAssayRunChange(assayDefinition?.protocolSchemaName);
+        navigate(AppURL.create(ASSAYS_KEY, assayProtocol.providerName, assayProtocol.name));
+    }, [assayDefinition?.protocolSchemaName, assayProtocol.name, assayProtocol.providerName, navigate]);
+
+    const hideConfirm = useCallback(() => {
+        setShowConfirmDelete(false);
+    }, []);
+
+    return (
+        <>
+            <CreatedModified row={runData}/>
+            {(allowReimport || allowDelete) && (
+                <ManageDropdownButton id="assay-run-details" pullRight={true} collapsed={true}>
+                    {allowReimport &&
+                        <AssayReimportRunButton runId={runId} replacedByRunId={runData?.ReplacedByRun?.value}/>
+                    }
+                    {allowDelete &&
+                        <DisableableMenuItem
+                            operationPermitted={canDelete}
+                            onClick={onDeleteRun}
+                            disabledMessage={getAssayRunDeleteMessage(canDelete, confirmationDataError)}
+                        >
+                            Delete Run
+                        </DisableableMenuItem>
+                    }
+                </ManageDropdownButton>
+            )}
+            {showConfirmDelete && (
+                <AssayRunDeleteModal
+                    afterDelete={afterDelete}
+                    afterDeleteFailure={hideConfirm}
+                    onCancel={hideConfirm}
+                    selectedRowId={runId}
+                />
+            )}
+        </>
+    );
+});
+
+export const AssayDesignHeaderButtons: FC<AssayHeaderButtonProps> = props => {
+    const {assayDefinition, assayProtocol, navigate, menuInit} = props;
+    const [showConfirmDeleteAssayDesign, setShowConfirmDeleteAssayDesign] = useState<boolean>(false);
+
+    const resetState = useCallback(() => {
+        setShowConfirmDeleteAssayDesign(false);
+    }, []);
+
+    const onDeleteAssayDesign = useCallback(() => {
+        setShowConfirmDeleteAssayDesign(true);
+    }, []);
+
+    const beforeAssayDesignDelete = useCallback(() => {
+        onAssayDesignChange(assayDefinition.protocolSchemaName);
+    }, [assayDefinition?.protocolSchemaName]);
+
+    const afterAssayDesignDelete = useCallback((success: boolean) => {
+        if (success) {
+            menuInit();
+            clearAssayDefinitionCache();
+            navigate(AppURL.create(ASSAYS_KEY));
+        } else {
+            // delay to make sure grid invalidate form beforeAssayDesignDelete is finished
+            window.setTimeout(resetState, 100);
+        }
+    }, [menuInit, navigate]);
+
+    return (
+        <>
+            <RequiresPermission permissionCheck="any"
+                                perms={[PermissionTypes.Read, PermissionTypes.DesignAssay, PermissionTypes.CanSeeAuditLog]}>
```

Comment:
which menu item is `Read` perm for here? is it the export design? Should this be `ReadAssay`?

Reply:
My take on it is that ReadAssay is controlling the reading of the data, but Read permission is enough to see the design of the assay, but server side it looks like we're checking for AssayReadPermission on the AssayDomainKind, so I'll update here.   This also makes me wonder if we aren't missing some checks for ReadAssay.  I take a look around.